patterns.csv

```diff
@@ -1,8 +1,8 @@
 "Pattern"	"Label"	"Description"
-"[A-Z0-9]{2}-[A-Z0-9]{2}-[A-Z0-9]{2}"	"MAT"	"Matriculas portuguesas 0A-0A-0A"
+"(?<=[^\w-])(?:[A-Z]{2}-\d{2}-\d{2}|[A-Z]{2}-\d{2}-[A-Z]{2}|\d{2}-[A-Z]{2}-\d{2}|\d{2}-\d{2}-[A-Z]{2}|[A-Z]{2}\s?\d{2}\s?[A-Z]{2})(?![A-Z\d])(?=[^\w-])"	"MAT"	"Matriculas portuguesas 0A-0A-0A"
 "\d+(\-\w+)?\/(\w+\.)(\w+\.?)+"	"PRO"	"Processos com / (NN/...) opcionalmente apanha (NN-W/...) até ao próximo espaço"
-"\b\d{1,2}(-|\.|/)\d{1,2}(-|\.|/)\d{2,4}\b"	"DAT"	"Datas NN.NN.NNNN ou NN-NN-NNNN ou NN/NN/NNNN"
+"0?[1-31]{1,2}[-\/][1-12]{1,2}[-\/][1-2023]{2,4}"	"DAT"	"Datas NN.NN.NNNN ou NN-NN-NNNN ou NN/NN/NNNN"
 "\d{4}-\d{3}\b"	"CEP"	"Código de endereçamento postal"
-"\S+@([\w-]+\.)+[\w-]{2,4}"	"E-MAIL"	"Endereços de correio eletrónico"
+"\S+@([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}(?:\.[a-zA-Z]{2,})?"	"E-MAIL"	"Endereços de correio eletrónico"
```

Comment:
What scenarios are failing that this will solve?

Reply:
The old regex only allowed TLDs of 2 to 4 characters, so it would fail to match ".travel", for example, which has 6 characters.
- user@domain.travel

Reply:
Why no simply remove the ",4}"?

Reply:
Because that way the rule would identify emails contains underscores ("_") which is invalid.
- Old rule: https://regexr.com/8166e
- Old rule without the "4": https://regexr.com/8166k
- New rule: https://regexr.com/8166h